CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+substantial commits, consider breaking it up into subtasks and mention them in the ticket's description. That way, you
+can create multiple pull requests that are easier to review. For example, break issue #28 into two subtasks and make
+a branch for each subtask, such as "GH-28-1" and "GH-28-2".
+
+<!-- TODO: Add an issue and pull request template on GitHub -->
+Go through the following steps to open a new issue/ticket:
+
+1. Search through the existing tickets to see if a similar ticket is already filed. If you find a ticket that seems to
+   match yours, add a comment to the same ticket describing any additional information you have (different context or
+   environment, speculated cause, possible solution, or any other feedback). Otherwise, create a new ticket.
+2. Provide as much relevant detail in the ticket description. If the ticket is a bug report, it should include the
+   following:
+   - Details of the actual behavior.
+   - Instructions for reproducing that bug.
+   - Difference between the actual and expected behavior.
+   - Additional details about the general context.
+3. Besides, more technical information can be added that might help whosoever working on that ticket (see below).
+   - Console logs or error messages.
+   - Suspected technical cause of the problem.
+   - Suggestions for resolving that bug.
+
+### Add Unit Tests
+
+Production-level code needs to be accompanied by a reasonable suite of unit tests. This helps others confirm that the
+code is working as intended, and allows the community to adopt more agile refactoring techniques while being more
+confident in our ability to avoid regressions. All unit tests should be written using jqUnit (check the
+[jqUnit documentation](https://docs.fluidproject.org/infusion/development/jqUnit.html) for more information).
+
+### Lint Your Code
+
+JavaScript is a highly dynamic and loose language, and many common errors are not picked up until run time. To avoid
+errors and common pitfalls in the language, and to maintain consistency in terms of syntax, lint your code regularly
+using the provided Grunt lint task (see below). Ensure that you run lint checks on your code before making a commit.
+
+``` snippet
+# Runs linting tasks
+grunt lint
+```
+
+### Create Meaningful Commit Logs
+
+All commits logs should include the following information:
+
+1. A reference to the GitHub issue this commit applies to (at the beginning of the first line).
+2. A short and meaningful summary of the commit, on the first line.
+3. A meaningful commit log describing the contents of the change. Also, mention if the pull request is meant to close a
+   ticket at the beginning. For example, "Resolves #18", "Fixes #28", et cetera.
+
+Sometimes, a commit may be trivial or entirely cosmetic (code reformatting, fixing typos in comments, et cetera).
+In those cases, it is acceptable to use the "NOGH:" prefix for your log. However, it is still required to provide a
```

Comment:
OK, I see what you're going for, you can omit the requirement for a ticket in the previous section.  I would improve the last sentence, something like, "However, you must still provide...".